### PR TITLE
Change latex command handling only for XML Chars

### PR DESCRIPTION
**Pull request by johannes-manner**
Refs #3838 
----

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes
- [X] Manually tested changed features in running JabRef

Tried to solve this issue in PR #4108.
This approach is now more dedicated to the XML docbook export and not applied to all fields as the first PR was.

I am also not sure, if this issue is related to #3781, #3644?

### CODE REVIEW

**Comment by johannes-manner** [comment]
@bernhard-kleine: I exported your provided bib entry via docbooks exporter into XML. Please check the result and let us know, if there is something missing.

[testBracesResult.xml.txt](https://github.com/JabRef/jabref/files/2487123/testBracesResult.xml.txt)
[testbraces.bib.txt](https://github.com/JabRef/jabref/files/2487124/testbraces.bib.txt)

I will write tests based on your files, if the export is now correct.

**Comment by bernhard-kleine** [comment]
This is a separate item from the docbook5 export where I am waiting for feedback.

**Comment by johannes-manner** [comment]
Added two test cases for (1) removed words based on an error during the latex command processing and for (2) unicode problems based on the same error.

File: src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        ExporterFactory exporterFactory = ExporterFactory.create(customFormats, layoutPreferences, savePreferences, xmpPreferences);
+
+        exportFormat = exporterFactory.getExporterByName("docbook").get();
+
+        databaseContext = new BibDatabaseContext();
+        charset = StandardCharsets.UTF_8;
+    }
+
+    @AfterEach
+    public void tearDown() {
```
Comment (LinusDietz) [inline src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java]: Is the tearDown method  really necessary?

File: src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+public class DocbookExporterTest {
+
+    public BibDatabaseContext databaseContext;
+    public Charset charset;
+
+    private Exporter exportFormat;
+
+    @BeforeEach
+    public void setUp() {
+        Map<String, TemplateExporter> customFormats = new HashMap<>();
```
Comment (LinusDietz) [inline src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java]: how about inlining `customFormats`?

File: src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        LayoutFormatterPreferences layoutPreferences = mock(LayoutFormatterPreferences.class, Answers.RETURNS_DEEP_STUBS);
+        SavePreferences savePreferences = mock(SavePreferences.class);
+        XmpPreferences xmpPreferences = mock(XmpPreferences.class);
+        ExporterFactory exporterFactory = ExporterFactory.create(customFormats, layoutPreferences, savePreferences, xmpPreferences);
+
+        exportFormat = exporterFactory.getExporterByName("docbook").get();
+
+        databaseContext = new BibDatabaseContext();
+        charset = StandardCharsets.UTF_8;
```
Comment (LinusDietz) [inline src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java]: You can move `databaseContext` and `charset` initialization into the field, so the `setUp` method is less complex.

**Comment by LinusDietz** [comment]
Ping. What is the status @johannes-manner?

**Comment by johannes-manner** [comment]
Oversee the comments and the ping. Sorry for that :)

**Pull request merged**